runtime/compiler/trj9/x/env/J9CPU.cpp

```diff
@@ -24,71 +24,85 @@
 #include "env/CompilerEnv.hpp"
 #include "env/CPU.hpp"
 #include "env/VMJ9.h"
+#include "x/runtime/X86Runtime.hpp"
 
 namespace J9
 {
 
 namespace X86
 {
 
-const char *
-CPU::getX86ProcessorVendorId(TR::Compilation *comp)
+TR_X86CPUIDBuffer *
+CPU::queryX86TargetCPUID()
    {
-   return comp->fej9()->getX86ProcessorVendorId();
+   static TR_X86CPUIDBuffer buf = { {'U','n','k','n','o','w','n','B','r','a','n','d'} };
+   jitGetCPUID(&buf);
+   return &buf;
    }
 
-uint32_t
-CPU::getX86ProcessorSignature(TR::Compilation *comp)
+const char *
+CPU::getX86ProcessorVendorId()
    {
-   return comp->fej9()->getX86ProcessorSignature();
+   static char buf[13];
+
+   // Terminate the vendor ID with NULL before returning.
+   //
+   strncpy(buf, self()->queryX86TargetCPUID()->_vendorId, 12);
+   buf[12] = '\0';
+
+   return buf;
    }
 
 uint32_t
-CPU::getX86ProcessorFeatureFlags(TR::Compilation *comp)
+CPU::getX86ProcessorSignature()
    {
-   return comp->fej9()->getX86ProcessorFeatureFlags();
+   return self()->queryX86TargetCPUID()->_processorSignature;
    }
 
 uint32_t
-CPU::getX86ProcessorFeatureFlags2(TR::Compilation *comp)
+CPU::getX86ProcessorFeatureFlags()
    {
-   return comp->fej9()->getX86ProcessorFeatureFlags2();
+   return self()->queryX86TargetCPUID()->_featureFlags;
    }
 
 uint32_t
-CPU::getX86ProcessorFeatureFlags8(TR::Compilation *comp)
+CPU::getX86ProcessorFeatureFlags2()
    {
-   return comp->fej9()->getX86ProcessorFeatureFlags8();
+   return self()->queryX86TargetCPUID()->_featureFlags2;
    }
 
-bool
-CPU::testOSForSSESupport(TR::Compilation *comp)
+uint32_t
+CPU::getX86ProcessorFeatureFlags8()
    {
-   return comp->fej9()->testOSForSSESupport();
+   return self()->queryX86TargetCPUID()->_featureFlags8;
    }
 
-void
-CPU::setX86ProcessorFeatureFlags(uint32_t flags)
+bool
+CPU::testOSForSSESupport()
    {
-   _featureFlags = flags;
+   return true; // VM guarantees SSE/SSE2 are available
```

Comment:
Does this mean that the VM will fail to start if SSE/SSE2 are not supported on the current processor?

Reply:
This code change was ported over from 
```
-bool
-TR_J9VMBase::testOSForSSESupport()
-   {
-   return true; // VM guarantees SSE/SSE2 are available
-   }
```
It will have the same behaviour as before. I don't have an answer for this question :(